### ISSUE TITLE
Generating JsonSchema from class with List<> property for Angular6-json-schema-form

### ISSUE THREAD

**Issue by lunchboxsushi** [RicoSuter/NJsonSchema#996]
```
public class Member {
   // this will be fine, it will be added to definitions and have the proper #ref mapping
   public Address Address {get; set} 

   // this will not be generated correctly (I have not found anything in to documentation 
     regarding Enumerable types
   public List<Phone> Phones {get; set;}
}
```

Currently I cannot find any documentation on how to handle List<> properties.

 the generated JsonSchema will look similar to the following

```
{ 
    $schema: "http://json-schema.org/draft-04/schema#",
    additionalProperties: false,
    definitions: {
        Address: {
            // will have props / additionalprops / type:object
        }
        // Missing Phone definition ?
    },
    properties: { 
       address: { #ref: '[path to reference]' },
       phones: { // this should be using #ref, what am I missing?
            items: { type: "string" },
            type: ["array", "null"],
        }
    }
}
```

I will get `formatFormData error: Schema type not found for form value at /Phone/0`

do I need to somehow tell the mapper this property is a reference to an object? 

My purpose is to create a JSON Schema based on my classes and pass it to the browser client that can render a form based on the schema standard using Angular6-Json-Schema.

**Comment by RicoSuter** [comment]
Do you map Person to string or what?

**Comment by lunchboxsushi** [comment]
Do you mean Phone? It should be mapped to another object

```
public class Phone {
    public int Id {get; set;}
    public string number: {get; set;}
}
```

**Comment by lunchboxsushi** [comment]
Also I am getting the same error as well for other properties that are `List<string>`

**Comment by RicoSuter** [comment]
It is by design for List<string>..
Can you provide a complete sample?

**Comment by lunchboxsushi** [comment]
```
// MODELS
 public class POSIntegrationOptions
    {      
        // remove properties that are fine
        public List<string> IgnorePaymentTypes { get; set; }
        public List<string> IgnoreCategories { get; set; }
        public List<string> IncludeOnlyCategories { get; set; }
        public List<POSIntegrationRegister> RegisterMap { get; set; }        
    }

public class CloverIntegrationOptions : POSIntegrationOptions, IPOSIntegartionOptions
    {
        // remove properties that are fine
        public POSScanTarget ScanTarget { get; set; }
    }
   
// Used to create a selector switch based on interface
public interface IPOSIntegartionOptions    {    }

---------------------------------------------------------------------

// HELPER : method to return base type on interface
 public static IPOSIntegartionOptions GetIntegrationOptionInstance(string integrationName)
        {
            switch (integrationName)
            {
                case "CLOVER":
                    return new CloverIntegrationOptions();                
                default:
                    return null;
            }
        }

// web request
 [HttpGet]
        public ActionResult GetIntegrationSettings(int Id)
        {
            var data = _integrationService.PopulateIntegrationViewModel(Id);
            var schemaSettings = new JsonSchemaGeneratorSettings
            {
                DefaultPropertyNameHandling = PropertyNameHandling.CamelCase,
                DefaultEnumHandling = EnumHandling.CamelCaseString,
                AllowReferencesWithProperties = true,
                FlattenInheritanceHierarchy = true, // Need this due to inheritance
                TypeMappers = { // integer not mappable to JSON object number
                    new PrimitiveTypeMapper(typeof(int), i => i.Type = JsonObjectType.Number)
                }
            };
            var instance = IntegrationManager.GetIntegrationOptionInstance(data.IntegrationType);

            var schemaString = JsonSchema.FromType(instance.GetType(), schemaSettings).ToJson();
            return SuccessResponse(JObject.Parse(schemaString));
        }

```

essentially I only have the issues with List<> types as they aren't generated correctly with #ref

**Comment by lunchboxsushi** [comment]
Here's the output json

```
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "CloverIntegrationOptions",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "ignorePaymentTypes": {
      "type": [
        "array",
        "null"
      ],
      "items": {
        "type": "string"
      }
    },
    "ignoreCategories": {
      "type": [
        "array",
        "null"
      ],
      "items": {
        "type": "string"
      }
    },
    "includeOnlyCategories": {
      "type": [
        "array",
        "null"
      ],
      "items": {
        "type": "string"
      }
    },
    "registerMap": {
      "type": [
        "array",
        "null"
      ],
      "items": {
        "$ref": "#/definitions/POSIntegrationRegister"
      }
    },
    "scanTarget": {
      "$ref": "#/definitions/POSScanTarget"
    }
  },
  "definitions": {
    "POSIntegrationRegister": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "registerId": {
          "type": [
            "null",
            "string"
          ]
        },
        "name": {
          "type": [
            "null",
            "string"
          ]
        },
        "deviceId": {
          "type": "number"
        }
      }
    },
    "POSScanTarget": {
      "type": "string",
      "description": "",
      "x-enumNames": [
        "Paired",
        "Self",
        "Both"
      ],
      "enum": [
        "paired",
        "self",
        "both"
      ]
    }
  }
}
```

**Comment by RicoSuter** [comment]
How do you get from this json schema to this console output?

**Comment by lunchboxsushi** [comment]
I was able to error create the output using anuglar6-json-schema library; However, I think I've found the culprit. when NJsonSchema generated a  `List<>` it will create `type: ["array", "null"]` which causes an issue with the angular library. I need figure out how to have all list type just create `type: "array"` without a required attribute

Sorry for the long discussion, everything has worked flawlessly so for I'm just having a hard time making the mappings are compatible between the two frameworks

The angular library will take the schema from NJsonSchema and upgrade it to supported version (6 or 7) of the draft, I wonder if this is possibly where the issue comes in.
